config/v1/types_dns.go

```diff
@@ -74,6 +80,16 @@ type DNSZone struct {
 	Tags map[string]string `json:"tags,omitempty"`
 }
 
+// Upstream defines the schema for associating a base domain with a list of name resolvers.
+type Upstream struct {
+	// domain is the base domain to match to forward name queries.
+	Domain string `json:"domain"`
```

Comment:
Should this have validation metadata and/or docs?

Are wildcards supported? If so, with what syntax?

Reply:
Must be a rfc 1123 subdomain.